From db9833c345d4b41f7e2e1004f44852c8839a9381 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:14:31 +0100
Subject: [PATCH 01/14] debian patch  00 -- packed.diff.

---
 usb.h.in | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/usb.h.in b/usb.h.in
index 443c7c0..7422b30 100644
--- a/usb.h.in
+++ b/usb.h.in
@@ -66,14 +66,14 @@
 struct usb_descriptor_header {
 	u_int8_t  bLength;
 	u_int8_t  bDescriptorType;
-};
+} __attribute__ ((packed));
 
 /* String descriptor */
 struct usb_string_descriptor {
 	u_int8_t  bLength;
 	u_int8_t  bDescriptorType;
 	u_int16_t wData[1];
-};
+} __attribute__ ((packed));
 
 /* HID descriptor */
 struct usb_hid_descriptor {
@@ -85,7 +85,7 @@ struct usb_hid_descriptor {
 	/* u_int8_t  bReportDescriptorType; */
 	/* u_int16_t wDescriptorLength; */
 	/* ... */
-};
+} __attribute__ ((packed));
 
 /* Endpoint descriptor */
 #define USB_MAXENDPOINTS	32
@@ -172,7 +172,7 @@ struct usb_device_descriptor {
 	u_int8_t  iProduct;
 	u_int8_t  iSerialNumber;
 	u_int8_t  bNumConfigurations;
-};
+} __attribute__ ((packed));
 
 struct usb_ctrl_setup {
 	u_int8_t  bRequestType;
@@ -180,7 +180,7 @@ struct usb_ctrl_setup {
 	u_int16_t wValue;
 	u_int16_t wIndex;
 	u_int16_t wLength;
-};
+} __attribute__ ((packed));
 
 /*
  * Standard requests
-- 
2.21.0

