From 275d621d66740737594788a7fc0219fbd8168721 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:15:38 +0100
Subject: [PATCH 03/14] debian patch  02 -- usbpp.diff

---
 usbpp.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usbpp.h b/usbpp.h
index abda64f..423b59d 100644
--- a/usbpp.h
+++ b/usbpp.h
@@ -54,7 +54,7 @@ namespace USB {
 		 * \returns the number of bytes sent, or a negative value on
 		 * failure
 		 */
-		int bulkWrite(QByteArray message, int timeout = 100);
+		int bulkWrite(unsigned char* message, int timeout = 100);
 		
 		/**
 		 * \brief Bulk read
-- 
2.21.0

