From f7a46ef6dd627ab0a2aeab5fcfb7845eb93ba4e6 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:16:36 +0100
Subject: [PATCH 05/14] debian patch  04 -- infinite loop.diff

---
 linux.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/linux.c b/linux.c
index 43cfaf8..a7128ea 100644
--- a/linux.c
+++ b/linux.c
@@ -213,15 +213,21 @@ static int usb_urb_transfer(usb_dev_handle *dev, int ep, int urbtype,
       return ret;
     }
 
-    FD_ZERO(&writefds);
-    FD_SET(dev->fd, &writefds);
-
 restart:
     waiting = 1;
     context = NULL;
     while (!urb.usercontext && ((ret = ioctl(dev->fd, IOCTL_USB_REAPURBNDELAY, &context)) == -1) && waiting) {
+      if (ret == -1)
+      {
+        if (errno == ENODEV)
+        {
+          return -ENODEV;
+        }
+      }
       tv.tv_sec = 0;
       tv.tv_usec = 1000; // 1 msec
+      FD_ZERO(&writefds);
+      FD_SET(dev->fd, &writefds);
       select(dev->fd + 1, NULL, &writefds, NULL, &tv); //sub second wait
 
       if (timeout) {
-- 
2.21.0

