From 9da98c51514ab1f95dd0c187fad404bdb747ed31 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:18:15 +0100
Subject: [PATCH 08/14] debian patch 07 -- altsetting_alloc.diff

---
 descriptors.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/descriptors.c b/descriptors.c
index 0146a82..ac1dd82 100644
--- a/descriptors.c
+++ b/descriptors.c
@@ -177,6 +177,7 @@ static int usb_parse_interface(struct usb_interface *interface,
     }
 
     ifp = interface->altsetting + interface->num_altsetting;
+    memset(ifp, 0, sizeof(struct usb_interface_descriptor));
     interface->num_altsetting++;
 
     usb_parse_descriptor(buffer, "bbbbbbbbb", ifp);
@@ -219,10 +220,7 @@ static int usb_parse_interface(struct usb_interface *interface,
     /* Copy any unknown descriptors into a storage area for */
     /*  drivers to later parse */
     len = (int)(buffer - begin);
-    if (!len) {
-      ifp->extra = NULL;
-      ifp->extralen = 0;
-    } else {
+    if (len) {
       ifp->extra = malloc(len);
       if (!ifp->extra) {
         if (usb_debug >= 1)
@@ -277,8 +275,7 @@ static int usb_parse_interface(struct usb_interface *interface,
         parsed += retval;
         size -= retval;
       }
-    } else
-      ifp->endpoint = NULL;
+    }
 
     /* We check to see if it's an alternate to this one */
     ifp = (struct usb_interface_descriptor *)buffer;
-- 
2.21.0

