From dc8358d0ab41b299a4503000213b60c5a91c7004 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:18:42 +0100
Subject: [PATCH 09/14] debian patch 08 bus_location.diff.

---
 linux.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/linux.c b/linux.c
index a7128ea..329c8f0 100644
--- a/linux.c
+++ b/linux.c
@@ -355,6 +355,7 @@ int usb_os_find_busses(struct usb_bus **busses)
 
     strncpy(bus->dirname, entry->d_name, sizeof(bus->dirname) - 1);
     bus->dirname[sizeof(bus->dirname) - 1] = 0;
+    bus->location = atoi(bus->dirname);
 
     LIST_ADD(fbus, bus);
 
-- 
2.21.0

