From 0757091a1249bbc0881169a11d400ada3f727208 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:19:05 +0100
Subject: [PATCH 10/14] debian patch 09 -- dummy.diff.

---
 Makefile.am  |  4 ++--
 configure.in | 17 +++++++++++++++--
 2 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index cab1309..57196e5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -17,7 +17,7 @@ libusb-config: libusb-config.in
 EXTRA_DIST = LICENSE libusb.spec.in libusb.spec libusb-config.in README.in README \
              INSTALL.libusb.in INSTALL.libusb Doxyfile apidocs/header.html \
              apidocs/footer.html apidocs/doxygen.css apidocs/doxygen.png libusb.pc.in
-EXTRA_libusb_la_SOURCE = linux.c linux.h bsd.c darwin.c
+EXTRA_libusb_la_SOURCE = linux.c linux.h bsd.c darwin.c dummy.c
 
 lib_LTLIBRARIES = libusb.la libusbpp.la
 
@@ -36,7 +36,7 @@ LDADDS = -Wl,-framework -Wl,IOKit -Wl,-framework -Wl,CoreFoundation -Wl,-prebind
 PREBIND_FLAGS=-Wl,-seg1addr,0x01666000
 PREBIND_FLAGSPP=-Wl,-seg1addr,0x01676000
 else
-OS_SUPPORT = 
+OS_SUPPORT = dummy.c
 endif
 endif
 endif
diff --git a/configure.in b/configure.in
index 331cec6..b0aed4e 100644
--- a/configure.in
+++ b/configure.in
@@ -11,6 +11,7 @@ AH_TOP(
 #undef LINUX_API
 #undef BSD_API
 #undef DARWIN_API
+#undef DUMMY_API
 
 #undef HAVE_OLD_DEV_USB_USB_H
 
@@ -89,6 +90,7 @@ dnl AC_CANONICAL_HOST
 LINUX_API=0
 DARWIN_API=0
 BSD_API=0
+DUMMY_API=0
 
 AC_MSG_CHECKING(for what USB OS support)
 case $host in
@@ -96,6 +98,7 @@ case $host in
     AC_DEFINE(LINUX_API, 1)
     AC_DEFINE(BSD_API, 0)
     AC_DEFINE(DARWIN_API, 0)
+    AC_DEFINE(DUMMY_API, 0)
     LINUX_API=1
     os_support=linux
     AC_MSG_RESULT(Linux)
@@ -105,6 +108,7 @@ case $host in
     AC_DEFINE(BSD_API, 1)
     AC_DEFINE(LINUX_API, 0)
     AC_DEFINE(DARWIN_API, 0)
+    AC_DEFINE(DUMMY_API, 0)
     BSD_API=1
     os_support=bsd
     AC_MSG_RESULT(FreeBSD, OpenBSD and/or NetBSD)
@@ -114,24 +118,33 @@ case $host in
     AC_DEFINE(DARWIN_API, 1) 
     AC_DEFINE(LINUX_API, 0)
     AC_DEFINE(BSD_API, 0)
+    AC_DEFINE(DUMMY_API, 0)
     DARWIN_API=1
     os_support=darwin
     AC_MSG_RESULT(Darwin and/or MacOS 10)
     OSLIBS="-Wl,-framework -Wl,IOKit -Wl,-framework -Wl,CoreFoundation -Wl,-prebind"
     ;;
   *)
-    AC_MSG_RESULT(unknown operating system)
-    AC_MSG_ERROR(libusb does not support compiling for $host)
+    AC_DEFINE(DARWIN_API, 0) 
+    AC_DEFINE(LINUX_API, 0)
+    AC_DEFINE(BSD_API, 0)
+    AC_DEFINE(DUMMY_API, 1)
+    DUMMY_API=1
+    os_support=dummy
+    AC_MSG_RESULT(unknown operating system $host)
+    OSLIBS=""
     ;;
 esac
 
 AC_SUBST(DARWIN_API)
 AC_SUBST(LINUX_API)
 AC_SUBST(BSD_API)
+AC_SUBST(DUMMY_API)
 
 AM_CONDITIONAL(LINUX_API, test "$os_support" = "linux")
 AM_CONDITIONAL(BSD_API, test "$os_support" = "bsd")
 AM_CONDITIONAL(DARWIN_API, test "$os_support" = "darwin")
+AM_CONDITIONAL(DUMMY_API, test "$os_support" = "dummy")
 
 AC_SUBST(OSLIBS)
 
-- 
2.21.0

