From 8e715be0539a4d9211a561f5a1f01740f295601e Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:19:58 +0100
Subject: [PATCH 12/14] debian patch 11 -- transfer timeout.diff.

---
 linux.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/linux.c b/linux.c
index 329c8f0..3b2b9a8 100644
--- a/linux.c
+++ b/linux.c
@@ -190,6 +190,8 @@ static int usb_urb_transfer(usb_dev_handle *dev, int ep, int urbtype,
     tv_ref.tv_sec++;
   }
 
+  waiting = 1;
+
   do {
     fd_set writefds;
 
@@ -214,7 +216,6 @@ static int usb_urb_transfer(usb_dev_handle *dev, int ep, int urbtype,
     }
 
 restart:
-    waiting = 1;
     context = NULL;
     while (!urb.usercontext && ((ret = ioctl(dev->fd, IOCTL_USB_REAPURBNDELAY, &context)) == -1) && waiting) {
       if (ret == -1)
@@ -243,6 +244,7 @@ restart:
     if (context && context != &urb) {
       context->usercontext = URB_USERCONTEXT_COOKIE;
       /* We need to restart since we got a successful URB, but not ours */
+      waiting = 1;
       goto restart;
     }
 
@@ -255,10 +257,10 @@ restart:
       USB_ERROR_STR(-errno, "error reaping URB: %s", strerror(errno));
 
     bytesdone += urb.actual_length;
-  } while ((ret == 0 || urb.usercontext) && bytesdone < size && urb.actual_length == requested);
+  } while ((ret == 0 || urb.usercontext) && bytesdone < size && urb.actual_length == requested && waiting);
 
   /* If the URB didn't complete in success or error, then let's unlink it */
-  if (ret < 0 && !urb.usercontext) {
+  if ((ret < 0 && !urb.usercontext) || (!waiting && bytesdone < size)) {
     int rc;
 
     if (!waiting)
-- 
2.21.0

