From 6e952601b4fc6834ccbf355abd5520e5a75e626b Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:20:54 +0100
Subject: [PATCH 13/14] debian patch 12 -- ac_prog_cxx.diff.

---
 configure.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.in b/configure.in
index b0aed4e..340facd 100644
--- a/configure.in
+++ b/configure.in
@@ -162,6 +162,7 @@ if test "x$enable_debug" = "xyes"; then
 fi
 
 # Checks for programs.
+AC_PROG_CXX
 AC_LANG_CPLUSPLUS
 AC_PROG_CC
 AM_PROG_CC_C_O
-- 
2.21.0

