From 78505b97d4fc82a769f434d0e51d364acfff7ec9 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 22 Feb 2020 11:50:17 +0100
Subject: [PATCH 14/14] linux.c, error: Fix snprintf buffer sizes.

---
 error.c |  2 +-
 error.h |  2 +-
 linux.c | 12 ++++++------
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/error.c b/error.c
index f7d496d..cc96b41 100644
--- a/error.c
+++ b/error.c
@@ -12,7 +12,7 @@
 #include "usb.h"
 #include "error.h"
 
-char usb_error_str[1024] = "";
+char usb_error_str[4*PATH_MAX] = "";
 int usb_error_errno = 0;
 usb_error_type_t usb_error_type = USB_ERROR_TYPE_NONE;
 
diff --git a/error.h b/error.h
index 173e6fd..38890f4 100644
--- a/error.h
+++ b/error.h
@@ -7,7 +7,7 @@ typedef enum {
   USB_ERROR_TYPE_ERRNO,
 } usb_error_type_t;
 
-extern char usb_error_str[1024];
+extern char usb_error_str[4*PATH_MAX];
 extern int usb_error_errno;
 extern usb_error_type_t usb_error_type;
 
diff --git a/linux.c b/linux.c
index 3b2b9a8..e4cc74b 100644
--- a/linux.c
+++ b/linux.c
@@ -22,10 +22,10 @@ static char usb_path[PATH_MAX + 1] = "";
 
 static int device_open(struct usb_device *dev)
 {
-  char filename[PATH_MAX + 1];
+  char filename[3*PATH_MAX + 4];
   int fd;
 
-  snprintf(filename, sizeof(filename) - 1, "%s/%s/%s",
+  snprintf(filename, 3*PATH_MAX + 3, "%s/%s/%s",
     usb_path, dev->bus->dirname, dev->filename);
 
   fd = open(filename, O_RDWR);
@@ -377,9 +377,9 @@ int usb_os_find_devices(struct usb_bus *bus, struct usb_device **devices)
   struct usb_device *fdev = NULL;
   DIR *dir;
   struct dirent *entry;
-  char dirpath[PATH_MAX + 1];
+  char dirpath[2*PATH_MAX + 3];
 
-  snprintf(dirpath, PATH_MAX, "%s/%s", usb_path, bus->dirname);
+  snprintf(dirpath, 2*PATH_MAX + 2, "%s/%s", usb_path, bus->dirname);
 
   dir = opendir(dirpath);
   if (!dir)
@@ -388,7 +388,7 @@ int usb_os_find_devices(struct usb_bus *bus, struct usb_device **devices)
 
   while ((entry = readdir(dir)) != NULL) {
     unsigned char device_desc[DEVICE_DESC_LENGTH];
-    char filename[PATH_MAX + 1];
+    char filename[3*PATH_MAX + 3];
     struct usb_device *dev;
     struct usb_connectinfo connectinfo;
     int i, fd, ret;
@@ -408,7 +408,7 @@ int usb_os_find_devices(struct usb_bus *bus, struct usb_device **devices)
     strncpy(dev->filename, entry->d_name, sizeof(dev->filename) - 1);
     dev->filename[sizeof(dev->filename) - 1] = 0;
 
-    snprintf(filename, sizeof(filename) - 1, "%s/%s", dirpath, entry->d_name);
+    snprintf(filename, 3*PATH_MAX + 2, "%s/%s", dirpath, entry->d_name);
     fd = open(filename, O_RDWR);
     if (fd < 0) {
       fd = open(filename, O_RDONLY);
-- 
2.21.0

